### DISCUSSION TITLE
GenericParameterSignature Name {!!0}

### DISCUSSION THREAD

**Discussion by dashne** [Washi1337/AsmResolver#442] [Ideas]
It will be nice to have real Generic Type (i.e. T ) instead of (!0 or !!0), when Calling Property
`GenericParameterSignature.Name`,Specially when using the signature and you do not have its 
Method's owner or Type, Like `ILSpy` or `dnSpy`

**Comment by ds5678** [comment]
I feel like that's a .NET limitation that has nothing to do with AsmResolver. I think that signature is just a byte and an int (or something like that). The name is unknowable, except in the context of a method.

**Comment by Washi1337** [comment]
I understand the thought process and it would indeed be nice, but this is something that I am not sure is a good idea to implement in AsmResolver directly. 

As @ds5678 indicated, `GenericParameterSignature` is defined to be a very small signature with only a flag indicating whether it was defined by a method or type, and the index of the parameter. Having to resolve them to actual generic parameter definitions would require two things:
- Hard-wiring the signature to a metadata member that is using the signature (to understand how the signature is used).
- Type resolution (e.g., to resolve a type or method reference to get to its definition and read out the original generic parameters).

Both points are problematic. Hard-wiring severely hurts the UX of blob signatures in general (e.g., it would disallow reuse of signatures by multiple metadata members, and having to pass the owner to a signature always clutters code etc.). Type resolution could be done but can be computationally expensive and is not even always guaranteed to succeed (e.g., in case of missing dependency files).
